Modsen.Server.Authentication.Api/Middlewares/ExceptionHandlerMiddleware.cs

```diff
@@ -0,0 +1,51 @@
+﻿using Modsen.Server.Authentication.Domain.Exceptions;
+using Newtonsoft.Json;
+
+namespace Modsen.Server.Authentication.Api.Middlewares
+{
+    public class ExceptionHandlerMiddleware
+    {
+        private readonly RequestDelegate _next;
+        public ExceptionHandlerMiddleware(RequestDelegate next)
+        {
+            _next = next;
+        }
+
+        public async Task Invoke(HttpContext context)
+        {
+            try
+            {
+                await _next.Invoke(context);
+            }
+            catch (Exception ex)
+            {
+                await HandleExceptionMessageAsync(context, ex).ConfigureAwait(false);
+            }
+        }
+
+        private static Task HandleExceptionMessageAsync(HttpContext context, Exception exception)
+        {
+            var statusCode = GetStatusCodeByException(exception);
+
+            var result = JsonConvert.SerializeObject(new
+            {
+                StatusCode = statusCode,
+                ErrorMessage = exception.Message
+            });
+            
+            context.Response.ContentType = "application/json";
+            context.Response.StatusCode = statusCode;
+            
+            return context.Response.WriteAsync(result);
+        }
+
+        private static int GetStatusCodeByException(Exception exception)
+        {
+            return exception switch
+            {
+                BadRequestException => 404,
```

Comment:
Bad request is 400. 404 is not found

Reply:
Also handle here NotFoundException. Add messages to your exceptions. Messages should be declared as constants